pkg/trace/agent/tags_test.go

```diff
@@ -148,7 +148,26 @@ func TestTagSetKey(t *testing.T) {
 
 func TestNormalizeTag(t *testing.T) {
 	for _, tt := range []struct{ in, out string }{
+		{in: "#test_starting_hash", out: "test_starting_hash"},
+		{in: "TestCAPSandSuch", out: "testcapsandsuch"},
+		{in: "Test Conversion Of Weird !@#$%^&**() Characters", out: "test_conversion_of_weird_characters"},
+		{in: "$#weird_starting", out: "weird_starting"},
+		{in: "allowed:c0l0ns", out: "allowed:c0l0ns"},
+		{in: "1love", out: "love"},
+		{in: "ünicöde", out: "ünicöde"},
+		{in: "ünicöde:metäl", out: "ünicöde:metäl"},
+		{in: "Data🐨dog🐶 繋がっ⛰てて", out: "data_dog_繋がっ_てて"},
+		{in: " spaces   ", out: "spaces"},
+		{in: " #hashtag!@#spaces #__<>#  ", out: "hashtag_spaces"},
+		{in: ":testing", out: ":testing"},
+		{in: "_foo", out: "foo"},
+		{in: ":::test", out: ":::test"},
+		{in: "contiguous_____underscores", out: "contiguous_underscores"},
+		{in: "foo_", out: "foo"},
+		{in: "", out: ""},
+		{in: " ", out: ""},
 		{in: "ok", out: "ok"},
+		{in: "™Ö™Ö™™Ö™", out: "ö_ö_ö"},
 		{in: "AlsO:ök", out: "also:ök"},
 		{in: ":still_ok", out: ":still_ok"},
 		{in: "___trim", out: "trim"},
```

Comment:
Can we also add a testcase with this interesting character we found? One case where it appears at the end and one case where it appears in the middle?

Reply:
👍 added one test case inside which it appears at the beginning, the middle and the end. It doesn't get touched by the algorithm because it is a valid Unicode letter.